janus_server/src/hpke.rs

```diff
@@ -144,38 +144,166 @@ impl HpkeSender {
+                    &self.recipient_config,
+                    application_info,
+                    plaintext,
+                    associated_data,
+                )
+            }
+            (HpkeKemId::X25519HkdfSha256, HpkeKdfId::HkdfSha256, HpkeAeadId::ChaCha20Poly1305) => {
+                seal::<ChaCha20Poly1305, HkdfSha256, X25519HkdfSha256>(
+                    &self.recipient_config,
+                    application_info,
+                    plaintext,
+                    associated_data,
+                )
+            }
+            (HpkeKemId::X25519HkdfSha256, HpkeKdfId::HkdfSha384, HpkeAeadId::Aes128Gcm) => {
+                seal::<AesGcm128, HkdfSha384, X25519HkdfSha256>(
+                    &self.recipient_config,
+                    application_info,
+                    plaintext,
+                    associated_data,
+                )
+            }
+            (HpkeKemId::X25519HkdfSha256, HpkeKdfId::HkdfSha384, HpkeAeadId::Aes256Gcm) => {
+                seal::<AesGcm256, HkdfSha384, X25519HkdfSha256>(
+                    &self.recipient_config,
+                    application_info,
+                    plaintext,
+                    associated_data,
+                )
+            }
+            (HpkeKemId::X25519HkdfSha256, HpkeKdfId::HkdfSha384, HpkeAeadId::ChaCha20Poly1305) => {
+                seal::<ChaCha20Poly1305, HkdfSha384, X25519HkdfSha256>(
+                    &self.recipient_config,
+                    application_info,
+                    plaintext,
+                    associated_data,
+                )
+            }
+            (HpkeKemId::X25519HkdfSha256, HpkeKdfId::HkdfSha512, HpkeAeadId::Aes128Gcm) => {
+                seal::<AesGcm128, HkdfSha512, X25519HkdfSha256>(
+                    &self.recipient_config,
+                    application_info,
+                    plaintext,
+                    associated_data,
+                )
+            }
+            (HpkeKemId::X25519HkdfSha256, HpkeKdfId::HkdfSha512, HpkeAeadId::Aes256Gcm) => {
+                seal::<AesGcm256, HkdfSha512, X25519HkdfSha256>(
+                    &self.recipient_config,
+                    application_info,
                     plaintext,
                     associated_data,
                 )
             }
             (HpkeKemId::X25519HkdfSha256, HpkeKdfId::HkdfSha512, HpkeAeadId::ChaCha20Poly1305) => {
                 seal::<ChaCha20Poly1305, HkdfSha512, X25519HkdfSha256>(
                     &self.recipient_config,
-                    HpkeApplicationInfo::new(
-                        self.task_id,
-                        self.label,
-                        self.sender_role,
-                        self.recipient_role,
-                    ),
+                    application_info,
                     plaintext,
                     associated_data,
                 )
             }
+            // This catch-all pattern is unreachable at time of writing, but it may be necessary
```

Comment:
WDYT about removing this case entirely (causing a compile failure if new enum values are ever added)?

This would force us to maintain compatibility with the version of the `hpke` crate we're using, since with this PR we're effectively opting into supporting everything `hpke` does.